### PR TITLE
Add new Jenkins job scripts for device-virtual-go

### PR DESCRIPTION
**Pull request by FelixTing**
Signed-off-by: Felix Ting <felix@iotechsys.com>

### CODE REVIEW

**Comment by jamesrgregg** [comment]
@MightyNerdEric I'm submitting a helpdesk ticket now to get @FelixTing set up with Jenkins Sandbox access so he can test this jjb tempalte first.

**Review by jamesrgregg — APPROVED** [review]
LGTM but it might need a view added unless device jobs are automatically added to that Jenkins view.
Verify job passes in Sandbox.

File: shell/edgexfoundry-device-virtual-go-docker-push.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# Ensure we fail the job if any steps fail
+set -e -o pipefail
+
+# Required parameters:
+#
+#  VERSION: Set this value based on the VERSION file in the project root.
+#  DEPLOY_TYPE: Can be `snapshot`, `staging` or`release`
```
Comment (eb-oss) [inline shell/edgexfoundry-device-virtual-go-docker-push.sh:8]: Where is DEPLOY_TYPE defined?

**Comment by eb-oss** [comment]
> LGTM but it might need a view added unless device jobs are automatically added to that Jenkins view.
> Verify job passes in Sandbox.

No, this doesn't need a view. There is a single view for all device jobs, defined here: https://github.com/FelixTing/ci-management/blob/add_device_virtual_go_jobs/jjb/device/device.yaml

File: shell/edgexfoundry-device-virtual-go-docker-push.sh (hunk 1 above)
Comment (FelixTing) [reply shell/edgexfoundry-device-virtual-go-docker-push.sh:8]: DEPLOY_TYPE is defined in [edgex-templates-go.yaml](https://github.com/edgexfoundry/ci-management/blob/master/jjb/edgex-templates-go.yaml#L168)

**Comment by jamesrgregg** [comment]
> > LGTM but it might need a view added unless device jobs are automatically added to that Jenkins view.
> > Verify job passes in Sandbox.
> 
> No, this doesn't need a view. There is a single view for all device jobs, defined here: https://github.com/FelixTing/ci-management/blob/add_device_virtual_go_jobs/jjb/device/device.yaml

Thanks @MightyNerdEric as soon as I typed that... I checked and noted I was mistaken.

**Comment by jamesrgregg** [comment]
the verify jobs are now passing for both arm and non-arm in Sandbox.  @MightyNerdEric please review once more when you get a chance.

**Comment by jamesrgregg** [comment]
@MightyNerdEric could you please take a look at this one. The jobs ran in the Sandbox.

**Review by eb-oss — APPROVED** [review]

**Pull request merged**